File: cpp/questions/longestConsecutive.cpp
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) {
            return 0;
        }
        std::unordered_set<int> hs(nums.begin(), nums.end());
        int ans = 0;
        for (int n: hs) {
            if (!hs.count(n-1)) {
                int cn = n;
                int cl = 1;
                while (hs.count(cn+1)) {
                    ++cn;
                    ++cl;
                }
                ans = std::max(ans, cl);
            }
        }
        return ans;
    }
};

File: typescript/questions/longestConsecutive.ts
function longestConsecutive(nums: number[]): number {
  if (nums.length <= 0) {
    return 0;
  }
  const st: Set<number> = new Set(nums);
  let ans = 1;
  for (const n of st) {
    if (!st.has(n - 1)) {
      let cn = n;
      let cl = 1;
      while (st.has(cn + 1)) {
        cn += 1;
        cl += 1;
      }
      ans = Math.max(ans, cl);
    }
  }
  return ans;
}
